### PR TITLE
PIBD segment p2p messages

### PR DESCRIPTION
**Pull request by jaspervdm**
This PR introduces the new p2p messages as defined in [this RFC](https://github.com/mimblewimble/grin-rfcs/pull/68). Only merge after https://github.com/mimblewimble/grin/pull/3482 and https://github.com/mimblewimble/grin/pull/3492. 
Until then we use a placeholder `Segmenter` that just panics on usage.

- ~Enforce segment height limits~
- ~Rebase once 3482 is merged~
- ~Switch to `BitmapSegment` in place of `Segment<BitmapChunk>` in `OutputBitmapSegment` responses~

### CODE REVIEW

**Comment by antiochp** [comment]
#3482 is now merged.

**Comment by antiochp** [comment]
:eyes:

File: p2p/src/protocol.rs [hunk 1]
```diff
@@ -287,6 +290,92 @@ impl MessageHandler for Protocol {
...
+					)?)
+				} else {
+					Consumed::None
+				}
+			}
+			Message::OutputBitmapSegment(_)
+			| Message::OutputSegment(_)
+			| Message::RangeProofSegment(_)
+			| Message::KernelSegment(_) => Consumed::None,
+
```
Comment (antiochp) [inline p2p/src/protocol.rs:378]: This is just a `/dev/null` placeholder for now?

File: servers/src/common/adapters.rs [hunk 2]
```diff
@@ -477,6 +482,66 @@ where
 	fn get_tmpfile_pathname(&self, tmpfile_name: String) -> PathBuf {
 		self.chain().get_tmpfile_pathname(tmpfile_name)
 	}
+
+	fn get_kernel_segment(
+		&self,
+		hash: Hash,
+		id: SegmentIdentifier,
+	) -> Result<Segment<TxKernel>, chain::Error> {
+		if id.height < 9 || id.height > 13 {
```
Comment (antiochp) [inline servers/src/common/adapters.rs:491]: We should pull these min and max heights out into `const` so its easier to see them all together.

**Review by antiochp — APPROVED** [review]
Looks good.
Minor comment about the min/max values.

File: p2p/src/protocol.rs (hunk 1 above)
Comment (jaspervdm) [reply p2p/src/protocol.rs:378]: Yes, nodes receiving segments just ignore them.

File: servers/src/common/adapters.rs (hunk 2 above)
Comment (jaspervdm) [reply servers/src/common/adapters.rs]: Done

File: servers/src/common/adapters.rs [hunk 3]
```diff
@@ -46,6 +46,12 @@ use crate::util::OneTime;
 use chrono::prelude::*;
 use chrono::Duration;
 use rand::prelude::*;
+use std::ops::Range;
+
+const KERNEL_SEGMENT_HEIGHT_RANGE: Range<u8> = 9..14;
```
Comment (antiochp) [inline servers/src/common/adapters.rs:51]: Nice. Didn't think of using ranges here. :+1:

**Pull request merged**